pkg/kubelet/server/metrics/metrics.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	kubeletSubsystem = "kubelet"
+)
+
+var (
+	// HTTPRequests tracks the number of the http requests received since the server started (excluding long-running requests)
+	HTTPRequests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: kubeletSubsystem,
+			Name:      "http_requests_total",
+			Help:      "Number of the http requests received since the server started (excluding long-running requests).",
+		},
+		// server_type aims to differentiate the readonly server and the readwrite server.
+		[]string{"method", "path", "host", "server_type"},
+	)
+	// HTTPRequestsDuration tracks the duration in seconds to serve http requests (excluding long-running requests)
+	HTTPRequestsDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Subsystem: kubeletSubsystem,
+			Name:      "http_requests_duration_seconds",
+			Help:      "Duration in seconds to serve http requests (excluding long-running requests).",
+			// Use DefBuckets for now, will customize the buckets if necessary.
+			Buckets: prometheus.DefBuckets,
+		},
+		[]string{"method", "path", "host", "server_type"},
+	)
+	// HTTPInflightRequests tracks the number of the http requests received in last seconds (excluding long-running requests)
+	HTTPInflightRequests = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: kubeletSubsystem,
+			Name:      "http_inflight_requests_gauge",
+			Help:      "Number of the http requests received in last seconds (excluding long-running requests).",
+		},
+		[]string{"server_type"},
+	)
+	// HTTPLongRunningRequests tracks the gauge of all active long-running http requests
+	HTTPLongRunningRequests = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: kubeletSubsystem,
+			Name:      "http_longrunning_requests_gauge",
```

Comment:
same comment as above.  Maybe `http_current_longrunning_requests`?

How is this different from inflight requests?

Reply:
This is determined in `isLongRunningRequest`:

```
	longRunningRequestPaths := []string{"exec", "attach", "portforward", "debug"}
```